cpp/tests/spatial/coordinate_transform_test.cu

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include <type_traits>
+#include <tests/utilities/base_fixture.hpp>
+#include <tests/utilities/column_utilities.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+#include <tests/utilities/cudf_gtest.hpp>
+#include <tests/utilities/type_lists.hpp>
+#include <cuspatial/coordinate_transform.hpp>
+
+using cudf::test::fixed_width_column_wrapper;
+
+template <typename T>
+struct ShiftTest : public cudf::test::BaseFixture {};
+
+TYPED_TEST_CASE(ShiftTest, cudf::test::FloatingPointTypes);
+   
+TYPED_TEST(ShiftTest, CoordinateTest)
```

Comment:
I also like to see tests of extreme cases, and error cases.

Extreme case: does this do what you expect when the origin of the output coordinates is at one of the poles?
Error case: does this correctly handle empty inputs? Non-floating-point inputs? Inputs with nulls?

Reply:
Can you help me understand why having a test with a larger number of elements is beneficial? If I were implementing a kernel manually, I can see that it would verify threading was working as intended for large numbers of inputs - but I'm using thrust under the hood, so that's taken care of. I can see how a benchmark could be useful, but the actual result of the function could be ignored (since we're testing performance not correctness). I'm hesitant to add a CPU implementation because I'm not confident the GPU implementation makes logical sense as-is. it feels like adding a CPU implementation increases a false sense of correctness. I'm not sure of the exact use case for lonlat => cartesian transformation, so it's difficult to tell what's useful / correct.